### PR TITLE
Add ansible collection viewset

### PR DESCRIPTION
**Pull request by daviddavis**
fixes #5520
https://pulp.plan.io/issues/5520

### CODE REVIEW

File: pulp_ansible/app/viewsets.py [hunk 1]
```diff
@@ -60,9 +63,33 @@ class RoleViewSet(SingleArtifactContentUploadViewSet):
     filterset_class = RoleFilter
 
 
+class CollectionFilter(BaseFilterSet):
+    """
+    FilterSet for Ansible CollectionVersions.
```
Comment (bmbouter) [inline pulp_ansible/app/viewsets.py]: s/CollectionVersions/Collection/ ?

File: pulp_ansible/app/viewsets.py [hunk 2]
```diff
@@ -60,9 +63,33 @@ class RoleViewSet(SingleArtifactContentUploadViewSet):
...
+        model = Collection
+        fields = ["namespace", "name"]
+
+
+class CollectionViewset(NamedModelViewSet, mixins.RetrieveModelMixin, mixins.ListModelMixin):
+    """
+    Viewset for Ansible Collections.
+    """
+
+    endpoint_name = "ansible/collections"
```
Comment (bmbouter) [inline pulp_ansible/app/viewsets.py]: Does this still nest under `/pulp/api/v3/` ?

**Comment by bmbouter** [comment]
Can you also rebuild the docs so that the `api.json` schema is generated. Otherwise this viewset won't be available in the rebuilt bindings.

File: pulp_ansible/app/viewsets.py (hunk 2 above)
Comment (daviddavis) [reply pulp_ansible/app/viewsets.py]: Yes

**Comment by daviddavis** [comment]
Yea, I can. Do we have to update api.json every time we modify the API? Is there any way it can be automated?

**Comment by bmbouter** [comment]
I share the desire for automation, but there isn't any that is simple and obvious to me. Travis doesn't author code changes in any way today, so we can't have it do it easily. The only thing I can think of is a git commit hook that grabs the file at commit time. What about something like that?

File: pulp_ansible/app/viewsets.py (hunk 2 above)
Comment (bmbouter) [reply pulp_ansible/app/viewsets.py]: Thanks!

**Comment by bmbouter** [comment]
Also I didn't answer your question: Yes we do have to update the api.json everytime.

**Comment by daviddavis** [comment]
I guess my question is why have api.json in version control instead of generating it at publish time?

**Comment by bmbouter** [comment]
It's a great question, I spent over a day trying to find a better way. To have this not be version control at docs-build time you would need a running Pulp to produce it for you. For `pulpcore` and `pulpcore-plugin` the docs are built on Travis, and in those two environments specifically there is a running Pulp so those don't need to `api.json` checked into version control.

The issue with plugins is that they don't have their docs built on Travis and pushed to `docs.pulpproject.org` instead all the plugins use readthedocs. RTD won't install a python package much less the multi-process setup of Pulp that would be needed. So the workaround is that the api.json is included in the docs themselves and RTD can build referring from that.

To fully solve the problem we'd have to have all docs start building on Travis and then we'd need to find a place to host them.

Any suggestions or ideas are welcome.

**Pull request merged**